dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/advanced/CacheFileMigrator.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.file.advanced
+
+import com.datadog.android.core.internal.persistence.file.FileHandler
+import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+import com.datadog.android.log.Logger
+import java.util.concurrent.ExecutorService
+
+internal class CacheFileMigrator(
+    private val fileHandler: FileHandler,
+    private val executorService: ExecutorService,
+    private val internalLogger: Logger
+) : DataMigrator<Boolean> {
+
+    // region DataMigrator
+
+    override fun migrateData(
+        previousState: Boolean?,
+        previousFileOrchestrator: FileOrchestrator,
+        newState: Boolean,
+        newFileOrchestrator: FileOrchestrator
+    ) {
+        if (!newState) return
```

Comment:
what is the meaning of `newState` with boolean type? From the call site in `SdkFeature` we have `newState` always `true`, so `if (!newState)` branch will never be executed.

Reply:
Yes basically this is due to the refactoring where I added a generic `previousState`/`newState` params. I used booleans as it's the most generic one, basically indeed if `newState` is false it means we don't want to perform the migration.